Training5/task1/src/redux/reducers/todo.js

```diff
@@ -0,0 +1,93 @@
+import { createReducer } from "@reduxjs/toolkit";
+import { createSelector } from "reselect";
+import {
+  createTodo,
+  createTodoSuccess,
+  createTodoFail,
+  fetchTodo,
+  fetchTodoFail,
+  fetchTodoSuccess,
+  editTodo,
+  editTodoFail,
+  editTodoSuccess,
+  removeTodo,
+  removeTodoSuccess,
+  removeTodoFail,
+} from "../actions/todo";
+
+const defaultActionStartFun = (state) => {
+  state.loading = true;
+  state.statusAction = false;
+  state.error = null;
+};
+
+const defaultActionErrorFun = (state, action) => {
+  state.loading = false;
+  state.error = action.payload;
+};
+
+const initialState = {
+  loading: false,
+  data: {},
+  statusAction: false,
+  error: null,
+};
+
+export const todoReducer = createReducer(initialState, {
+  //Fetch
+  [fetchTodo]: (state) => {
+    state.loading = true;
+  },
+  [fetchTodoSuccess]: (state, action) => {
+    state.loading = false;
+    state.data = action.payload;
+  },
+  [fetchTodoFail]: defaultActionErrorFun,
+
+  //Create
+  [createTodo]: defaultActionStartFun,
+  [createTodoSuccess]: (state) => {
+    state.loading = false;
+    state.statusAction = true;
+  },
+  [createTodoFail]: defaultActionErrorFun,
+
+  //Edit
+  [editTodo]: defaultActionStartFun,
+  [editTodoSuccess]: (state) => {
+    state.loading = false;
+    state.statusAction = true;
+  },
+  [editTodoFail]: defaultActionErrorFun,
+
+  //Remove
+  [removeTodo]: defaultActionStartFun,
+  [removeTodoSuccess]: (state) => {
+    state.loading = false;
+    state.statusAction = true;
+  },
+  [removeTodoFail]: defaultActionErrorFun,
+});
+
+export const selectorTodoByNameCompleteStatus = createSelector(
```

Comment:
@thienvu23 you can't pass parameters to the selector created by createSelector
https://github.com/reduxjs/reselect#q-how-do-i-create-a-selector-that-takes-an-argument
you can move searchNameValue & completed into the redux store or using react props https://github.com/reduxjs/reselect#accessing-react-props-in-selectors

Reply:
@vanbui1995 Yes, I'll fix it soon.